library/L0_Platform/msp432p401r/msp432p401r.h

```diff
@@ -0,0 +1,1291 @@
+* Available Peripherals                                                       *
+******************************************************************************/
+#define __MCU_HAS_ADC14__     /**< Module ADC14 is available */
+#define __MCU_HAS_AES256__    /**< Module AES256 is available */
+#define __MCU_HAS_CAPTIO0__   /**< Module CAPTIO0 is available */
+#define __MCU_HAS_CAPTIO1__   /**< Module CAPTIO1 is available */
+#define __MCU_HAS_COMP_E0__   /**< Module COMP_E0 is available */
+#define __MCU_HAS_COMP_E1__   /**< Module COMP_E1 is available */
+#define __MCU_HAS_CRC32__     /**< Module CRC32 is available */
+#define __MCU_HAS_CS__        /**< Module CS is available */
+#define __MCU_HAS_DIO__       /**< Module DIO is available */
+#define __MCU_HAS_DMA__       /**< Module DMA is available */
+#define __MCU_HAS_EUSCI_A0__  /**< Module EUSCI_A0 is available */
+#define __MCU_HAS_EUSCI_A1__  /**< Module EUSCI_A1 is available */
+#define __MCU_HAS_EUSCI_A2__  /**< Module EUSCI_A2 is available */
+#define __MCU_HAS_EUSCI_A3__  /**< Module EUSCI_A3 is available */
+#define __MCU_HAS_EUSCI_B0__  /**< Module EUSCI_B0 is available */
+#define __MCU_HAS_EUSCI_B1__  /**< Module EUSCI_B1 is available */
+#define __MCU_HAS_EUSCI_B2__  /**< Module EUSCI_B2 is available */
+#define __MCU_HAS_EUSCI_B3__  /**< Module EUSCI_B3 is available */
+#define __MCU_HAS_FLCTL__     /**< Module FLCTL is available */
+#define __MCU_HAS_PCM__       /**< Module PCM is available */
+#define __MCU_HAS_PMAP__      /**< Module PMAP is available */
+#define __MCU_HAS_PSS__       /**< Module PSS is available */
+#define __MCU_HAS_REF_A__     /**< Module REF_A is available */
+#define __MCU_HAS_RSTCTL__    /**< Module RSTCTL is available */
+#define __MCU_HAS_RTC_C__     /**< Module RTC_C is available */
+#define __MCU_HAS_SYSCTL__    /**< Module SYSCTL is available */
+#define __MCU_HAS_TIMER32__   /**< Module TIMER32 is available */
+#define __MCU_HAS_TIMER_A0__  /**< Module TIMER_A0 is available */
+#define __MCU_HAS_TIMER_A1__  /**< Module TIMER_A1 is available */
+#define __MCU_HAS_TIMER_A2__  /**< Module TIMER_A2 is available */
+#define __MCU_HAS_TIMER_A3__  /**< Module TIMER_A3 is available */
+#define __MCU_HAS_TLV__       /**< Module TLV is available */
+#define __MCU_HAS_WDT_A__     /**< Module WDT_A is available */
+
+/* Definitions to show that specific ports are available */
+
+#define __MSP432_HAS_PORTA_R__
+#define __MSP432_HAS_PORTB_R__
+#define __MSP432_HAS_PORTC_R__
+#define __MSP432_HAS_PORTD_R__
+#define __MSP432_HAS_PORTE_R__
+#define __MSP432_HAS_PORTJ_R__
+
+#define __MSP432_HAS_PORT1_R__
+#define __MSP432_HAS_PORT2_R__
+#define __MSP432_HAS_PORT3_R__
+#define __MSP432_HAS_PORT4_R__
+#define __MSP432_HAS_PORT5_R__
+#define __MSP432_HAS_PORT6_R__
+#define __MSP432_HAS_PORT7_R__
+#define __MSP432_HAS_PORT8_R__
+#define __MSP432_HAS_PORT9_R__
+#define __MSP432_HAS_PORT10_R__
```

Comment:
These may need to be deleted at some point if they conflict with other header definitions.